### PR TITLE
feat(manager/azure-pipelines): Deployment jobs support

### PR DESCRIPTION
**Pull request by jabbera**
## Changes

Azure Devops has the concept of deployment jobs that are not covered by rennovate. This PR adds support for these jobs. I'm open to another approach for testing these changes as the # of fixture has exploded.

Microsoft Docs on Deployment Jobs: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/deployment-jobs?view=azure-devops#runonce-deployment-strategy 

## Context

Explained above

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/azure-pipelines/__fixtures__/azure-pipelines-canary.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+jobs:
```
Comment (viceice) [inline lib/modules/manager/azure-pipelines/__fixtures__/azure-pipelines-canary.yaml:1]: please inline those fixtures

**Review by viceice — COMMENTED** [review]
please convert types to zod schema validation

**Comment by jabbera** [comment]
> please convert types to zod schema validation

This was a rabbit hole. I raised some comments where I made changes and why they were made.

File: lib/modules/manager/azure-pipelines/extract.ts [hunk 2]
```diff
@@ -104,15 +109,62 @@ export function parseAzurePipelines(
...
+}
+
+function extractJob(job?: Job): PackageDependency<Record<string, any>>[] {
+  return extractSteps(job?.steps);
+}
 
-  return pkg;
+function extractDeploy(
+  deploy?: Deploy,
+): PackageDependency<Record<string, any>>[] {
```
Comment (viceice) [inline lib/modules/manager/azure-pipelines/extract.ts:143]:
```suggestion
function extractDeploy(
  deploy: Deploy | undefined,
): PackageDependency[] {
```

**Comment by jabbera** [comment]
Hi. This PR now needs to be rebased. @viceice should I wait until you comment on my above question or rebase now?

**Review by rarkins — DISMISSED** [review]
Needs de-conflicting

**Review by viceice — APPROVED** [review]

**Comment by jabbera** [comment]
Thanks all!

**Pull request merged**